packages/frontend/src/components/RefreshServer/RefreshStepsModal/index.tsx

```diff
@@ -0,0 +1,165 @@
+                    icon: 'warning-sign',
+                    status: 'Error',
+                };
+            case 'RUNNING':
+                return {
+                    title: 'Sync in progress',
+                    icon: 'refresh',
+                    status: 'In progress',
+                };
+
+            case 'PENDING':
+                return {
+                    title: 'Sync in progress',
+                    icon: 'refresh',
+                    status: 'Queued',
+                };
+
+            default:
+                return {
+                    title: 'Sync in progress',
+                    icon: 'refresh',
+                    status: 'Success',
+                };
+        }
+    };
+
+    const totalSteps = refreshStepsMockup.steps.length;
+    const numberOfCompletedSteps = refreshStepsMockup.steps.filter((step) => {
+        return step.stepStatus === 'DONE';
+    });
+
+    return (
+        <Drawer
+            autoFocus
+            canEscapeKeyClose
+            canOutsideClickClose
+            enforceFocus
+            hasBackdrop
+            isCloseButtonShown
+            isOpen={isOpen}
+            onClose={() => closeModal(false)}
+            shouldReturnFocusOnClose
+            size={'400px'}
+            title={
+                <RefreshStepsHeadingWrapper className={Classes.DIALOG_HEADER}>
+                    <Icon
+                        icon={statusInfo(refreshStepsMockup.jobStatus).icon}
+                        size={18}
+                    />
+                    <div>
+                        <RefreshStepsTitle>
+                            {statusInfo(refreshStepsMockup.jobStatus).title}
+                        </RefreshStepsTitle>
+                        <StepsCompletionOverview>{`${numberOfCompletedSteps.length}/${totalSteps} steps complete - 00:01:30`}</StepsCompletionOverview>
+                    </div>
+                </RefreshStepsHeadingWrapper>
+            }
+            position={Position.RIGHT}
+        >
+            <StepsWrapper>
+                {refreshStepsMockup.steps.map((step) => (
+                    <Step status={step.stepStatus}>
+                        <div>
+                            {step.stepStatus !== 'PENDING' && (
+                                <StepIcon
+                                    icon={statusInfo(step.stepStatus).icon}
+                                    status={step.stepStatus}
+                                />
+                            )}
+                        </div>
+                        <div>
+                            <StepName>{step.name}</StepName>
+                            <StepStatusWrapper>
+                                <StepStatus status={step.stepStatus}>
+                                    {statusInfo(step.stepStatus).status}{' '}
+                                </StepStatus>
+                                {!step.error && step.runningTime}
+                            </StepStatusWrapper>
```

Comment:
Should we hide the running time when is pending as well ? or will the runningTime be undefined ?

Reply:
It should be undefined at this point